### PR TITLE
Allow specific session keys to bypass reset_session

### PR DESCRIPTION
**Pull request by grega**
Adds `session_keys_to_persist` config option in order to carry over specific session keys from pre-log in to post-log in

(since `reset_session` is called post-log in in order to prevent session fixation: https://guides.rubyonrails.org/security.html#session-fixation-countermeasures)

### CODE REVIEW

**Comment by grega** [comment]
`spec/dummy/spec/requests/auth_request_spec.rb:185` is failing due to the session apparently being reset (🤷‍♂️) between the initial GET and the first POST, test `puts` debug output below (see change in `session_id` and the dropping of the `foo` key):

```
Initial session
{"session_id"=>"74f435ac474f4ddf31e27df9350e184f", "foo"=>"bar"}

/auth/rpi session
{"session_id"=>"aa9118694c9f56bfcde0e90427ea8a09", "omniauth.params"=>{}}

/rpi_auth/auth/callback session: 
{"session_id"=>"a93bb0d8ddb3695fb0858f0f77f95056", "current_user"=>{"user_id"=>"3ed9b57a-1eb9-42e1-ae54-9a11c930f035", "country"=>nil, "country_code"=>"GB", "email"=>"person@example.com", "email_verified"=>nil, "name"=>"Sylvester McBean", "nickname"=>"Sylvie", "picture"=>"https://placecage.com/100/100", "postcode"=>nil, "profile"=>"https://my.raspberry.pi/profile/edit", "roles"=>nil}}
```

**Comment by grega** [comment]
Moved CI / Bundle faff to: https://github.com/RaspberryPiFoundation/rpi-auth/pull/66

**Comment by patch0** [comment]
This all looks fine to me :)

File: README.md [hunk 1]
```diff
@@ -33,6 +33,7 @@ RpiAuth.configure do |config|
   config.identity_url = 'http://localhost:3002'        # The url for the profile instance being used for auth
   config.user_model = 'User'                           # The name of the user model in the host app being used, use the name as a string, not the model itself
   config.scope = 'openid email profile force-consent'  # The required OIDC scopes
+  config.session_keys_to_persist = 'foo bar'           # Optional: any session keys to persist across sessions (as the session is reset upon log in)
```
Comment (ce-rpf) [inline README.md:36]: Does this need to be a string or could we use an array here, that would avoid the need for the `split` when processing it?

Comment (grega) [reply README.md:36]: Yes good point, my thinking behind string was:

* It matches the input format used elsewhere, such as `config.scope`
* If taking config values from env vars, we should assume string (though, I'm not convinced we would usually want to utilise env vars for `config.session_keys_to_persist` values, more likely it's going to set in the initializer and will be same for all envs)

Comment (ce-rpf) [reply README.md:36]: I see that the test uses an array here so would be good to have the example and the test consistent too?

Comment (grega) [reply README.md:36]: Good spot, yep fixing...

**Pull request merged**